src/transform.ts

```diff
@@ -8,24 +8,81 @@
  */
 
 import * as R from 'ramda';
-import { Left, Right, Either } from 'monads.either';
+
+/**
+ * This class encapsulates either an object of type L (the 'Left'), or
+ * an object of type R (the 'Right'), but not both at the same time,
+ * and never neither of the two.
+ *
+ * Obviously, this class belongs in some type-level util library,
+ * but until that lib exists, this code is the only place it's used.
+ */
+export class Either<L, R> {
```

Comment:
Definitely agree.
I played around with that for a while; I guess my thinking was colored by the simplicity of the Haskell version of this -- ie., there's no need for a discriminant, since values always know how they were constructed.